mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/ChessFacade.kt

```diff
@@ -82,6 +86,48 @@ class ChessFacade(private val auth: Auth, private val store: Store) {
       it.filterNotNull().mapNotNull(ChessDocument::uid).map { uid -> StoreMatch(uid, store) }
     }
   }
+
+  /**
+   * Determines the [MatchResult] of a given [Match]
+   *
+   * @param match the [Match] to determine its [MatchResult]
+   */
+  suspend fun determineMatchOutcome(match: Match): MatchResult? {
+    val game = match.game.filterNotNull().first()
+    when (game.nextStep) {
+      is NextStep.Checkmate -> MatchResult.Reason.CHECKMATE
+      is NextStep.Stalemate -> Tie
+      else -> Tie
+    }
+    return Tie
+  }
```

Comment:
Please check with @barmettlerl how this can be done nicely. Also, `MatchResult` is a view concept, and should therefore not be used in the `ChessFacade`.

Reply:
Yes, you can merge my branch into yours